packages/dev/core/src/Animations/animatable.core.ts

```diff
@@ -0,0 +1,1113 @@
+                        const scale = runtimeAnimation.weight / normalizer;
+
+                        if (!scale) {
+                            continue;
+                        } else if (runtimeAnimation.currentValue.scaleAndAddToRef) {
+                            runtimeAnimation.currentValue.scaleAndAddToRef(scale, finalValue);
+                        } else {
+                            finalValue += runtimeAnimation.currentValue * scale;
+                        }
+                    }
+
+                    // Add up the additive animations
+                    for (let animIndex = 0; animIndex < holder.additiveAnimations.length; animIndex++) {
+                        const runtimeAnimation = holder.additiveAnimations[animIndex];
+                        const scale: number = runtimeAnimation.weight;
+
+                        if (!scale) {
+                            continue;
+                        } else if (runtimeAnimation.currentValue.scaleAndAddToRef) {
+                            runtimeAnimation.currentValue.scaleAndAddToRef(scale, finalValue);
+                        } else {
+                            finalValue += runtimeAnimation.currentValue * scale;
+                        }
+                    }
+                }
+            }
+            target[path] = finalValue;
+        }
+
+        target._lateAnimationHolders = {};
+    }
+    scene._registeredForLateAnimationBindings.reset();
+}
+
+/** @internal */
+export function RegisterTargetForLateAnimationBinding(scene: Scene, runtimeAnimation: RuntimeAnimation, originalValue: any): void {
+    const target = runtimeAnimation.target;
+    scene._registeredForLateAnimationBindings.pushNoDuplicate(target);
+
+    if (!target._lateAnimationHolders) {
+        target._lateAnimationHolders = {};
+    }
+
+    if (!target._lateAnimationHolders[runtimeAnimation.targetPath]) {
+        target._lateAnimationHolders[runtimeAnimation.targetPath] = {
+            totalWeight: 0,
+            totalAdditiveWeight: 0,
+            animations: [],
+            additiveAnimations: [],
+            originalValue: originalValue,
+        };
+    }
+
+    if (runtimeAnimation.isAdditive) {
+        target._lateAnimationHolders[runtimeAnimation.targetPath].additiveAnimations.push(runtimeAnimation);
+        target._lateAnimationHolders[runtimeAnimation.targetPath].totalAdditiveWeight += runtimeAnimation.weight;
+    } else {
+        target._lateAnimationHolders[runtimeAnimation.targetPath].animations.push(runtimeAnimation);
+        target._lateAnimationHolders[runtimeAnimation.targetPath].totalWeight += runtimeAnimation.weight;
+    }
+}
+
+/**
+ * Initialize all the inter dependecies between the animations and Scene and Bone
+ * @param scenePrototype defines the scene prototype to use
+ * @param bonePrototype defines the bone prototype to use
+ */
+export function InitAnimations(scenePrototype: Scene, bonePrototype: Bone): void {

```

Comment:
I think the intention would be more clear if this was:
```suggestion
export function InitAnimations(sceneClass: typeof Scene, boneClass: typeof Bone): void {
```

And then usage would just be `InitAnimations(Scene, Bone)`
Otherwise you could pass a `Scene` or `Bone` instance, which I don't think is the intention (unless we think it is useful to be able to scope this to an instance).

Reply:
Good point